### PR TITLE
V3

### PR DESCRIPTION
**Pull request by weilandia**
3.0.0
- [BREAKING CHANGE] Improves format of text extracts to better support post processing into sentences/rows. This will only cause issues if you have tests/code that relies on the specific format of whitespace characters in the resulting extracts.
- Improves memory allocation for xlsx files using `Roo#each_row_streaming`
- Adds github actions as ci
- Simplifies private api away from individual "format extractors"

### CODE REVIEW

File: lib/simple_text_extract.rb [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
   def self.extract(filename: nil, raw: nil, filepath: nil, tempfile: nil)
-    TextExtractor.new(filename: filename, raw: raw, filepath: filepath, tempfile: tempfile).to_s
+    Extract.new(filename:, raw:, filepath:, tempfile:).to_s
   end
 
   def self.supports?(filename: nil)
     SUPPORTED_FILETYPES.include?(filename.split(".").last)
   end
+
+  def self.missing_dependency?(command)
```
Comment (damwhit) [inline lib/simple_text_extract.rb:19]: Nice added check here

File: test/simple_text_extract_test.rb [hunk 2]
```diff
@@ -83,22 +83,22 @@ def test_it_parses_pdf_files_to_text_from_tempfile
   end
 
   def test_it_parses_xlsx_files_to_text_from_path
-    assert_includes SimpleTextExtract.extract(filepath: "test/fixtures/test_xlsx.xlsx"), "Sheet1 ruby 25 Sheet2 js 35"
+    assert_includes SimpleTextExtract.extract(filepath: "test/fixtures/test_xlsx.xlsx"), "Sheet1\nruby 25\nSheet2\njs 35"
```
Comment (damwhit) [inline test/simple_text_extract_test.rb:86]: Do we want to omit special chars here? I feel like it makes sense to be consistent on that across filetypes. Ie. now docx files omit them and xlsx does not.

**Review by damwhit — COMMENTED** [review]
Nice job on these changes! Left one actionable comment to consider.

File: test/simple_text_extract_test.rb (hunk 2 above)
Comment (weilandia) [reply test/simple_text_extract_test.rb:86]: Good callout, I updated the docx parser to preserve newlines as well

**Review by damwhit — APPROVED** [review]
I like the change. LGTM

**Pull request merged**